### PR TITLE
Make postgrex respect env var PGDATABASE like libpq.

### PR DESCRIPTION
**Pull request by scouten**
I'd like to be able to configure my database name in production via environment variable and this would help. (FWIW I'm an Elixir and Ecto newbie. Happy to make changes if needed.)

### CODE REVIEW

**Comment by scouten** [comment]
Shamelessly modeled after #11.

File: test/login_test.exs [hunk 1]
```diff
@@ -97,6 +97,14 @@ defmodule LoginTest do
     assert {:ok, %Postgrex.Result{}} = P.query(pid, "SELECT 123", [])
   end
 
+  test "env var default db name" do
+    System.put_env("PGDATABASE", "postgrex_test")
+    opts = []
```
Comment (fishcakez) [inline test/login_test.exs]: `sync_connect: true` please so `start_link` blocks to connect

File: test/login_test.exs [hunk 2]
```diff
@@ -97,6 +97,14 @@ defmodule LoginTest do
...
+    assert {:ok, pid} = P.start_link(opts)
+    assert {:ok, %Postgrex.Result{}} = P.query(pid, "SELECT 123", [])
+    System.delete_env("PGDATABASE")
```
Comment (fishcakez) [inline test/login_test.exs]: If the test fails we won't reset, can you copy the dance done with `try do..after.. end` and `PGPORT` later in this file?

**Comment by scouten** [comment]
@fishcakez: Sure thing. It may take me a day or two to get to it, but happy to make the changes.

**Comment by fishcakez** [comment]
@scouten this is a feature you want so do as you wish :smile. It took a week for this to get reviewed so we were very slow on this end, sorry.

**Comment by scouten** [comment]
@fishcakez: I've updated per your suggestions. Please re-review. Thanks!

File: test/login_test.exs [hunk 3]
```diff
@@ -97,6 +97,42 @@ defmodule LoginTest do
...
+
+        assert_receive {:EXIT, _, {%Postgrex.Error{postgres: %{code: :invalid_catalog_name}}, [_|_]}}
+      end
+
+      capture_log fn ->
+        opts = [ backoff_type: :stop ]
+        {:ok, pid} = P.start_link(opts)
+        assert_receive {:EXIT, ^pid, {%Postgrex.Error{postgres: %{code: :invalid_catalog_name}}, [_|_]}}
+      end
+    after
```
Comment (fishcakez) [inline test/login_test.exs]: We only need to test this once, either with or without `:sync_connect`. I think the `:sync_connect example is clearer as we would only needs to assert on the `start_link` return value. No need to check for the `:EXIT`.

**Comment by scouten** [comment]
@fishcakez: Thanks for the quick review. Just pushed another update.

**Comment by fishcakez** [comment]
Thank you :heart:

**Pull request merged**